packages/babel-generator/src/generators/types.ts

```diff
@@ -1,7 +1,14 @@
 import type Printer from "../printer";
-import { isAssignmentPattern, isIdentifier } from "@babel/types";
+import {
+  isAssignmentPattern,
+  isIdentifier,
+  isJSXAttribute,
+} from "@babel/types";
 import type * as t from "@babel/types";
+import type { NodePath } from "@babel/traverse";
 import jsesc from "jsesc";
+// eslint-disable-next-line import/extensions
+import { escapeUTF8 as htmlEncode } from "entities/lib/escape.js";
```

Comment:
`escapeUTF8` escapes `'` as `&apos;`, which is an HTML 5 addition that is [not allowed by JSX](https://facebook.github.io/jsx/#prod-HTMLNamedCharacterReferenceName).

I think `escapeAttribute` is the right function?

Reply:
Thanks!

Reply:
This didn't contain `<>`, nothing seemed to fit, so I copy-pasted the code.

Reply:
Inside a double-quoted attribute, escaping `<>` doesn’t hurt but isn’t strictly necessary. The parser is only looking for entity references and a terminating `"`.